### PR TITLE
Customizeable milk limit

### PR DESCRIPTION
**Pull request by ghost**
These patches allow MME to track and set actor-specific values for the maximum milk limit. Currently all actors are initialized with the same values, but these can be modified via the MCM/Debug MilkMaid menu.

Something similar can be implemented for other (currently) fixed values and then used to enhance the roleplaying aspect. (Orcs are more resistant to pain, Khajiit generate more milk, Elves have lower storage capacity, ...)

### CODE REVIEW

File: 00_Core/Scripts/Source/MilkQUEST.psc [hunk 1]
```diff
@@ -558,8 +558,10 @@ Function MilkCycle(Actor akActor, int t)
 		StorageUtil.AdjustFloatValue(akActor,"MME.MilkMaid.TimesMilked", MilkTick)
 		MaidLevelCheck(akActor)
 	EndIf
+	; ^--- the maid's / slave's level (and the maximum milk amount) may have increased now
+	; (but the script continues with the old values - unsure whether this is intended or a bug)
```
Comment (ak86) [inline 00_Core/Scripts/Source/MilkQUEST.psc:562]: bug

File: 00_Core/Scripts/Source/MilkQUEST.psc [hunk 2]
```diff
@@ -571,16 +573,19 @@ Function MilkCycle(Actor akActor, int t)
...
 		else 
 			MilkCnt = MilkTick
 		endif
 	else
-		if MilkTick > ((MaidLevel+2)*2) && PiercingCheck(akActor) != 2
-			MilkTick = MilkTick - 1 * MilkTick / ((MaidLevel+2)*2)
+		if MilkTick > MilkMax && PiercingCheck(akActor) != 2
+			MilkTick = MilkTick - 1 * MilkTick / MilkMax
+			; suspicious calculation, please review
+			;  variant a) MilkTick - MilkTick / MilkMax
```
Comment (ak86) [inline 00_Core/Scripts/Source/MilkQUEST.psc:587]: lets make it

```
if MilkTick > MilkMax && PiercingCheck(akActor) != 2
    if BreastScaleLimit 
        MilkTick = MilkMax
    else
        MilkTick = MilkTick - (MilkTick * GushPct/100) - MilkTick / MilkMax 
    endif
    AddMilkFx(akActor, 1)
    AddLeak(akActor)
endif
```

MilkCnt = MilkTick

File: 00_Core/Scripts/Source/MilkQUEST.psc [hunk 3]
```diff
@@ -594,9 +599,18 @@ Function MilkCycle(Actor akActor, int t)
 
 	StorageUtil.SetFloatValue(akActor,"MME.MilkMaid.LactacidCount", LactacidCnt)
 	StorageUtil.SetFloatValue(akActor,"MME.MilkMaid.MilkGen", MaidMilkGen)
-	StorageUtil.SetFloatValue(akActor,"MME.MilkMaid.MilkCount", MilkCnt)
+	MME_Storage.setMilkCurrent(akActor, MilkCnt)
 	StorageUtil.SetFloatValue(akActor,"MME.MilkMaid.PainCount", PainCnt)
 
+	; setMilkCurrent() automatically restricts the provided value to the maximum allowed value
```
Comment (ak86) [inline 00_Core/Scripts/Source/MilkQUEST.psc:605]: the problem is  theres an option that allows breasts to hold more than max allowed milk, and its on by default

File: 00_Core/Scripts/Source/MilkQUEST.psc (hunk 2 above)
Comment (ghost) [inline 00_Core/Scripts/Source/MilkQUEST.psc:587]: Let me think about this a bit more. 'setMilkCurrent()' and related functions also need to be adjusted to support 'no Limit' properly.

File: 00_Core/Scripts/Source/MilkQUEST.psc (hunk 3 above)
Comment (ghost) [inline 00_Core/Scripts/Source/MilkQUEST.psc:605]: Sorry, I totally forgot about this. I can add this to the patch set and update the pull request.

On a more general note: Do you like the idea of having actor-specific/customizable limits?

Comment (ak86) [inline 00_Core/Scripts/Source/MilkQUEST.psc:605]: sure

Comment (ghost) [inline 00_Core/Scripts/Source/MilkQUEST.psc:605]: Can you help me out with https://github.com/ThomasNeumann/Milk-Mod-Economy/commit/33b48a64f2ce9236ff8751be7e367c2cd50e575b ? I'm having trouble to reference a property from MilkQuest. I added the following lines to be able to access the global limit

```
MilkQUEST Property MilkQ Auto
[...]
return MilkQ.BreastScaleLimit
```

but PapyrusCompiler complains

> MME_Storage.psc(175,8): variable MilkQ is undefined
> MME_Storage.psc(175,14): none is not a known user-defined type

I tried to do exactly as in MilkMCM.psc, but something seems to be missing.

Comment (ak86) [inline 00_Core/Scripts/Source/MilkQUEST.psc:605]: erm idk, for me it compiles

**Pull request closed without merging**